### PR TITLE
Add testing files for all classes within Task package

### LINKED ISSUE #141: Write test cases for Task related classes under Model

### PR DESCRIPTION
**Pull request by charliemoweng**
Fix #141

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 1]
```diff
@@ -58,10 +58,14 @@
 
     public static final String VALID_TASK_NAME_0 = "Assignment 1";
     public static final String VALID_TASK_NAME_1 = "Lab 7";
+    public static final String VALID_TASK_NAME_2 = "Assignment3";
     public static final String VALID_TASK_ID_0 = "T1";
     public static final String VALID_TASK_ID_1 = "T77";
+    public static final String VALID_TASK_ID_2 = "T333";
     public static final String VALID_TASK_DEADLINE_0 = "2021-10-28";
```
Comment (Ruofan1023) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:65]: Did we change the task ddl from DD-MM-YYYY to YYYY-MM-DD?

This is the current error message for adding task

image

File: src/test/java/seedu/address/testutil/TaskBuilder.java [hunk 2]
```diff
@@ -82,8 +83,14 @@ public TaskBuilder withStatus(Boolean isComplete) {
...
 
+    /**
+     * Build a Task based on the fields specified.
+     * @return The Task object built.
+     */
     public Task build() {
-        return new Task(moduleName, taskId, taskName, taskDeadline);
+        return this.isComplete
+                ? new Task(moduleName, taskId, taskName, taskDeadline, isComplete)
+                : new Task(moduleName, taskId, taskName, taskDeadline);
```
Comment (Ruofan1023) [inline src/test/java/seedu/address/testutil/TaskBuilder.java:93]: Is this the same as return `new Task(moduleName, taskId, taskName, taskDeadline, this.isComplete)` ?

**Comment by WangGLJoseph** [comment]
I think all the deadline related test cases will need to be changed to align with the java Date object that we're gonna be using

**Review by tanjoewel — APPROVED** [review]
LGTM!
Nothing to add on besides what others have already brought up

File: src/test/java/seedu/address/testutil/TaskBuilder.java (hunk 2 above)
Comment (charliemoweng) [reply src/test/java/seedu/address/testutil/TaskBuilder.java:93]: Yes i realise they are the same. thanks!

**Review by WangGLJoseph — APPROVED** [review]
LGTM! If possible, we could also try changing some of the valid deadlines to include the time as well (hh:mm) in the future

File: src/test/java/seedu/address/model/task/TaskTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        assertTrue(TASK1.isSameTask(TASK1));
+
+        // null -> returns false
+        assertFalse(TASK1.isSameTask(null));
+
+        // different task -> return false
+        assertFalse(TASK1.isSameTask(TASK2));
+
+        // same module but different ID -> return false
+        Task taskDifferentId = new TaskBuilder().withModule(MODULE_NAME_0).withId("T0")
```
[the remark this answers is not in the document]
Comment (charliemoweng) [reply src/test/java/seedu/address/model/task/TaskTest.java:40]: Changed! Thanks for the headsup

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 1 above)
Comment (charliemoweng) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java:65]: yes. updated task deadline format to reflect our latest change.

File: src/test/java/seedu/address/model/task/TaskDeadlineTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Test
+    public void constructor_invalidDeadline_throwsIllegalArgumentException() {
+        String invalidDeadline = " ";
+        assertThrows(IllegalArgumentException.class, () -> new TaskDeadline(invalidDeadline));
+    }
+
+    @Test
+    void isValidTaskDeadline() {
+        // only contains alphanumeric characters -> returns true
```
[the remark this answers is not in the document]
Comment (charliemoweng) [reply src/test/java/seedu/address/model/task/TaskDeadlineTest.java:32]: changed all these outdated formats to new format

**Pull request merged**